app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/masterpasswordgenerator/MasterPasswordGeneratorScreen.kt

```diff
@@ -0,0 +1,227 @@
+                            MasterPasswordGeneratorAction.PreventLockoutClickAction,
+                        )
+                    }
+                },
+                modifier = Modifier.standardHorizontalMargin(),
+            )
+            Spacer(modifier = Modifier.navigationBarsPadding())
+        }
+    }
+}
+
+@Composable
+private fun MasterPasswordGeneratorContent(
+    generatedPassword: String,
+    onGenerateNewPassword: () -> Unit,
+    onLearnToPreventLockout: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier.fillMaxWidth(),
+    ) {
+        Spacer(modifier = Modifier.height(24.dp))
+        BitwardenTextField(
+            label = "",
+            value = generatedPassword,
+            onValueChange = {},
+            readOnly = true,
+            shouldAddCustomLineBreaks = true,
+            textStyle = LocalNonMaterialTypography.current.sensitiveInfoSmall,
+            visualTransformation = nonLetterColorVisualTransformation(),
+            modifier = Modifier.fillMaxWidth(),
+        )
+
+        Spacer(modifier = Modifier.height(12.dp))
+        BitwardenFilledButtonWithIcon(
+            label = stringResource(R.string.generate_button_label),
+            onClick = onGenerateNewPassword,
+            icon = rememberVectorPainter(id = R.drawable.ic_generator),
+            modifier = Modifier.fillMaxWidth(),
+        )
+        Spacer(modifier = Modifier.height(12.dp))
+        Text(
+            text = stringResource(R.string.write_this_password_down_and_keep_it_somewhere_safe),
+            style = MaterialTheme.typography.bodySmall,
+            color = MaterialTheme.colorScheme.onSurfaceVariant,
+        )
+        BitwardenClickableText(
+            label = stringResource(R.string.learn_about_other_ways_to_prevent_account_lockout),
+            style = MaterialTheme.typography.labelMedium,
+            color = MaterialTheme.colorScheme.primary,
+            onClick = onLearnToPreventLockout,
+        )
+    }
+}
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class)
+private fun MasterPasswordGeneratorTopBar(
+    scrollBehavior: TopAppBarScrollBehavior,
+    onBackClick: () -> Unit,
+    onSaveClick: () -> Unit,
+) {
+    BitwardenTopAppBar(
+        title = stringResource(R.string.generate_master_password),
+        scrollBehavior = scrollBehavior,
+        navigationIcon = rememberVectorPainter(id = R.drawable.ic_back),
+        navigationIconContentDescription = stringResource(id = R.string.back),
+        onNavigationIconClick = onBackClick,
+        actions = {
+            BitwardenTextButton(
+                label = stringResource(id = R.string.save),
+                labelTextColor = MaterialTheme.colorScheme.primary,
+                onClick = onSaveClick,
+            )
+        },
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Preview
+@Composable
+private fun MasterPasswordGeneratorTopBarPreview() {
+    BitwardenTheme {
+        Surface {
```

Comment:
Why do we need the surface here?

Reply:
nothing...and actually dont need it on the other either since I forgot about `showBackground`.